File: src/main/java/net/moddingplayground/onastick/api/screen/LocalGrindstoneScreenHandler.java
package net.moddingplayground.onastick.api.screen;

import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.entity.player.PlayerInventory;
import net.minecraft.item.Item;
import net.minecraft.screen.GrindstoneScreenHandler;
import net.minecraft.screen.ScreenHandlerContext;

import java.util.Set;

public class LocalGrindstoneScreenHandler extends GrindstoneScreenHandler {
    protected final Set<Item> set;

    public LocalGrindstoneScreenHandler(int syncId, PlayerInventory inventory, ScreenHandlerContext context, Item... items) {
        super(syncId, inventory, context);
        this.set = Set.of(items);
    }

    @Override
    public boolean canUse(PlayerEntity player) {
        return player.getInventory().containsAny(this.set);
    }
}

File: src/main/kotlin/dev/andante/onastick/screen/LocalGrindstoneScreenHandler.kt
package dev.andante.onastick.screen

import net.minecraft.entity.player.PlayerEntity
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.item.Item
import net.minecraft.screen.GrindstoneScreenHandler
import net.minecraft.screen.ScreenHandlerContext

class LocalGrindstoneScreenHandler(
    syncId: Int,
    inventory: PlayerInventory,
    context: ScreenHandlerContext,
    vararg items: Item
) : GrindstoneScreenHandler(syncId, inventory, context) {
    private val set = setOf(*items)

    override fun canUse(player: PlayerEntity): Boolean {
        return player.inventory.containsAny(set)
    }
}
